winlogbeat/checkpoint/checkpoint.go

```diff
@@ -89,6 +89,12 @@ func NewCheckpoint(file string, maxUpdates int, interval time.Duration) (*Checkp
 		}
 	}
 
+	// Write the state file to verify we have have permissions.
+	err = c.flush()
```

Comment:
anything written to file before flush? Have you tested this working? Not sure on my side.